engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/HashCodeNode.java

```diff
@@ -616,7 +616,15 @@ long hashCodeForNull(Object selfNull, @CachedLibrary("selfNull") InteropLibrary
     return 0;
   }
 
-  @Specialization(guards = "isJavaObject(hostObject)")
+  @Specialization(
+      guards = {
+        "isJavaObject(hostObject)",
+        "!interop.hasMembers(hostObject)",
+        "!interop.hasArrayElements(hostObject)",
+        "!interop.isTime(hostObject)",
+        "!interop.isDate(hostObject)",
+        "!interop.isTimeZone(hostObject)"
+      })
   long hashCodeForHostObject(
```

Comment:
After adding few new small `java.math.BigInteger` values to `ValuesGenerator`, this specialization started to trigger for a `HostObject` around `java.util.List` - e.g. something that should go to `hasArrayElement` specialization.

Adding few _negative guards_, but possibly the right thing is to use the [SpecializationType pattern](https://github.com/enso-org/enso/blob/fc557f8fd2804379e4dad52b44d484b5aa36b748/engine/runtime/src/main/java/org/enso/interpreter/node/callable/resolver/HostMethodCallNode.java#L30).